### ISSUE TITLE
04_DataVirtualization - azuredbexternaltable notebook feedback.

### ISSUE THREAD

**Issue by jeroenterheerdt** [microsoft/sqlworkshops#148]
- please make it clearer that step 0 needs to be executed against Azure SQL.
- please make it clearer that step 1 onwards needs to be executed against SQL 2019
- step 3 failed for me with 
`- Error message
Started executing query at Line 17
Msg 46530, Level 16, State 11, Line 1
External data sources are not supported with type GENERIC.
Total execution time: 00:00:00`

although Polybase was installed and the services were running. I had to run the follow command and restart the SQL instance to make the command work. I know this issue was reported https://github.com/microsoft/sqlworkshops/issues/57 as well, but I am certain in this case Polybase was installed and services were running.

**Comment by jeroenterheerdt** [comment]
@rgward

**Comment by rgward** [comment]
Jeroen,

Thanks for filing this. Here is my response to your issues:

please make it clearer that step 0 needs to be executed against Azure SQL.

Step 0 literally says "Create a database in Azure SQL called wwiazure. Execute the following T-SQL to create a table and insert a row in the database". I'm not sure how much more clear I can make this.

please make it clearer that step 1 onwards needs to be executed against SQL 2019

The notebook is for Polybase for SQL Server 2019. What other data source would they possibly choose to execute the notebook against?

step 3 failed for me with
- Error message Started executing query at Line 17 Msg 46530, Level 16, State 11, Line 1 External data sources are not supported with type GENERIC. Total execution time: 00:00:00

There is no way to get this error message unless the PB feature is not installed and enabled. When you say this " I had to run the follow command and restart the SQL instance to make the command work.". What command are you referring to. I'm happy to dig into this but I don't know how to reproduce this unless I don't install and enable Polybase

**Comment by jeroenterheerdt** [comment]
Hi Bob,

- step 0: that assumes that someone actually knows what they are doing. What if someone has never created a Azure SQL database? They have not seen it in this workshop yet - so I would suggest at least providing a link to some documentation.
- step 1: precisely my point - just because there is a context switch to Azure SQL although we are in a Polybase for SQL Server 2019 notebook it makes sense to just add a note like: 'make sure to run this against your SQL Server 2019 instance'. That does not hurt anyone right?

- step 3: and yet I did. I had PB installed and the services were running. What I had to do is to enable the PB feature and restart the instance by running: `exec sp_configure @configname = 'polybase enabled', @configvalue = 1;` as was mentioned here https://github.com/Microsoft/azuredatastudio/issues/2819. 
I just ran the install wizard to make sure PB was installed, checked that the PB services were running in configuration manager. I did not do anything else and then the error appeared. So either I missed a step or this is an edge case where PB actually does not get enabled. I would suggest adding this fix to the notebook - but it's your call.

Sorry for all opening all the issues here - I am just going through everything one by one and trying to view things without assumptions and prior knowledge. It might seem as if there are only complaints - but rest assured - I checked everything and these are the only things that I was not sure of.
